frontend/src/component/executiveDashboard/componentsChart/MetricsSummaryChart/MetricsSummaryChart.tsx

```diff
@@ -1,36 +1,118 @@
 import type { ExecutiveSummarySchema } from 'openapi';
-import { LineChart, NotEnoughData } from '../../components/LineChart/LineChart';
+import {
+    fillGradientPrimary,
+    LineChart,
+    NotEnoughData,
+} from '../../components/LineChart/LineChart';
 import { MetricsSummaryTooltip } from './MetricsChartTooltip/MetricsChartTooltip';
 import { useMetricsSummary } from '../../hooks/useMetricsSummary';
 import { usePlaceholderData } from 'component/executiveDashboard/hooks/usePlaceholderData';
 import type { GroupedDataByProject } from '../../hooks/useGroupedProjectTrends';
+import { useTheme } from '@mui/material';
 
 interface IMetricsSummaryChartProps {
     metricsSummaryTrends: GroupedDataByProject<
         ExecutiveSummarySchema['metricsSummaryTrends']
     >;
+    isAggregate?: boolean;
+}
+
+function aggregateDataPerDate(
+    items: ExecutiveSummarySchema['metricsSummaryTrends'],
+) {
+    return items.reduce(
+        (acc, item) => {
+            if (!acc[item.date]) {
+                acc[item.date] = {
+                    totalApps: 0,
+                    totalEnvironments: 0,
+                    totalFlags: 0,
+                    totalNo: 0,
+                    totalRequests: 0,
+                    totalYes: 0,
+                };
+            }
+
+            acc[item.date].totalApps += item.totalApps;
+            acc[item.date].totalEnvironments += item.totalEnvironments;
+            acc[item.date].totalFlags += item.totalFlags;
+            acc[item.date].totalNo += item.totalNo;
+            acc[item.date].totalRequests += item.totalRequests;
+            acc[item.date].totalYes += item.totalYes;
+
+            return acc;
+        },
+        {} as {
+            [date: string]: {
+                totalApps: number;
+                totalEnvironments: number;
+                totalFlags: number;
+                totalNo: number;
+                totalRequests: number;
+                totalYes: number;
+            };
+        },
+    );
 }
 
 export const MetricsSummaryChart: VFC<IMetricsSummaryChartProps> = ({
     metricsSummaryTrends,
+    isAggregate,
 }) => {
-    const data = useMetricsSummary(metricsSummaryTrends);
+    const theme = useTheme();
+    const metricsSummary = useMetricsSummary(metricsSummaryTrends);
     const notEnoughData = useMemo(
-        () => !data.datasets.some((d) => d.data.length > 1),
-        [data],
+        () => !metricsSummary.datasets.some((d) => d.data.length > 1),
+        [metricsSummary],
     );
     const placeholderData = usePlaceholderData();
 
+    const aggregatedPerDay = useMemo(() => {
+        const result = aggregateDataPerDate(
+            Object.values(metricsSummary.datasets).flatMap((item) => item.data),
+        );
+        const data = Object.entries(result)
+            .map(([date, trends]) => ({ date, ...trends }))
+            .sort(
+                (a, b) =>
+                    new Date(a.date).getTime() - new Date(b.date).getTime(),
+            );
```

Comment:
Should we get some tests for this? Feels very logic heavy to not get some tests up.